vidgear/gears/streamgear.py

```diff
@@ -810,7 +810,10 @@ def __generate_hls_stream(self, input_params, output_params):
 
         # Finally, some hardcoded HLS parameters (Refer FFmpeg docs for more info.)
         output_params["-allowed_extensions"] = "ALL"
-        output_params["-hls_segment_filename"] = "{}-stream-%03d.{}".format(
+        # Normally output_params.get("stream_count", 0) would be used to access a dict variable,
+        # but in this case stream_count is always defined by default with 0
+        segment_template = "{}-stream%v-%03d.{}" if output_params["stream_count"] > 1 else "{}-stream-%03d.{}"
```

Comment:
@Vboivin  logic is wrong, `output_params["stream_count"] > 1` must be `output_params["stream_count"] > 0` instead.